app/helpers/event_helper.rb

```diff
@@ -0,0 +1,194 @@
+      content_tag(:p, @event.creator.username) +
+      content_tag(:p, @event.description) +
+      content_tag(:p, time_tag(@event.date)) +
+      content_tag(:div, class: 'text-right') do
+        if current_user == @event.creator
+
+          (link_to 'Edit', edit_event_path) + tag('br') +
+            (link_to 'Delete', deletee_event_path(@event.id), method: :delete,
+                                                              data: { confirm: 'Delete this event?' })
+
+        elsif @event.attendees.include?(current_user)
+
+          (link_to 'Remove event', cancel_visitor_event_path)
+
+        else
+
+          (link_to 'Want to Join --- Go ', visitor_event_path unless @event.date < Time.now)
+
+        end
+      end
+  end
+
+  def all_events_links(events)
+    all_contents = content_tag(:br)
+    events.each do |event|
+      content = content_tag(:tr, '')
+      content << content_tag(:td, event.creator.username)
+      content << content_tag(:td, event.name)
+      content << content_tag(:td, event.description)
+      content << content_tag(:td, event.date)
+      content << content_tag(:td, (link_to 'See Event', event_path(event.id)))
+      content << content_tag(:tr)
+      all_contents << content
+    end
+    all_contents
+  end
+
+  def upcoming_events
+    all_contents = content_tag(:br)
+    @upcoming_events = Event.upcoming
+    @upcoming_events.each do |event|
+      content = content_tag(:tr, '')
+      content << content_tag(:td, event.creator.username)
+      content << content_tag(:td, event.name)
+      content << content_tag(:td, event.description)
+      content << content_tag(:td, event.date)
+      all_contents << content
+    end
+    all_contents
+  end
+
+  def past_events
+    all_contents = content_tag(:br)
+    @past_events = Event.past
+    @past_events.each do |event|
+      content = content_tag(:td, event.creator.username)
+      content << content_tag(:td, event.name)
+      content << content_tag(:td, event.description)
+      content << content_tag(:td, event.date)
+      content << content_tag(:tr)
+      all_contents << content
+    end
+    all_contents
+  end
+
+  def user_upcoming_events(user)
+    all_contents = content_tag(:tr)
+    user.attended_events.upcoming.each do |event|
+      content = content_tag(:tr, '')
+      content << content_tag(:td, event.name)
+      content << content_tag(:td, event.description)
+      content << content_tag(:td, event.date)
+      content << content_tag(:br)
+      all_contents << content
+    end
+    all_contents
+  end
+
+  def user_past_events(user)
+    all_contents = content_tag(:tr)
+    user.attended_events.past.each do |event|
+      content = content_tag(:tr, '')
+      content << content_tag(:td, event.name)
+      content << content_tag(:td, event.description)
+      content << content_tag(:td, event.date)
+      content << content_tag(:br)
+      all_contents << content
+    end
+    all_contents
+  end
```

Comment:
You are repeating yourself and creating some content in each method. You can easily refactor the creation of content in a method. Let that method take dynamic input and you can pass in a hash with values. This way you won't be repeating yourself and you will only have to worry about one place while maintaining this particular part of your application.